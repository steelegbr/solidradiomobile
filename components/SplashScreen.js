import React from 'react';
import { Component } from 'react';
import { StyleSheet, View, ActivityIndicator, Image } from 'react-native'
import { Button, Text } from 'react-native-paper';
import { connect } from 'react-redux';
import { initialLoad } from '../reducers/actions';

class SplashScreen extends Component {

    render() {
        const { loadingStatus, initialLoad } = this.props;

        if (loadingStatus == 'not_started' || loadingStatus == 'started' || loadingStatus == 'success') {

            return(
                <View style={splashStyles.parent}>
                    <Image 
                        style={splashStyles.image}
                        source={require('../assets/Logo.png')}
                        resizeMode="contain"
                    />
                    <ActivityIndicator
                        style={splashStyles.spinner}
                        size="large"
                    />
                    <Text
                        style={splashStyles.text}
                    >
                        We won't be a moment...
                    </Text>
                </View>
            );

        } else {

            return(
                <View style={splashStyles.parent}>
                    <Image 
                        style={splashStyles.image}
                        source={require('../assets/Logo.png')}
                        resizeMode="contain"
                    />
                    <Text
                        style={splashStyles.text}
                    >
                        Something went wrong trying to get a classic with every byte.
                    </Text>
                    <Button
                        mode="contained"
                        color="#FFFFFF"                    
                        style={splashStyles.button}
                        onPress={() => {initialLoad()}}
                    >
                        Try Again!
                    </Button>
                </View>
            );

        }
    }

}

const splashStyles = StyleSheet.create({
    parent: {
        flex: 1,
        alignItems: "stretch",
        flexDirection: "column",
        backgroundColor: "#7300AE",
        justifyContent: "center",
        alignItems: "center"
    },
    text: {
        flex: 0,
        width: "100%",
        color: "white",
        fontSize: 20,
        fontWeight: "bold",
        textAlign: "center",
        padding: 20
    },
    spinner: {
        flex: 1
    },
    image: {
        flex: 3,
        width: "80%",
        height: undefined
    },
    button: {
        margin: 20
    }
});

const mapStateToProps = state => {
    return {
        loadingStatus: state.initialLoad
    };
};

const mapDispatchToProps = {
    initialLoad
}

export default connect(mapStateToProps, mapDispatchToProps)(SplashScreen);