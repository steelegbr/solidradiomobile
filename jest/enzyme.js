/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import 'react-native';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

/**
 * Set up Enzyme to mount to DOM, simulate events,
 * and inspect the DOM in tests.
 */
Enzyme.configure({ adapter: new Adapter() });

// JSDOM - required for Enzyme's mount

const { JSDOM } = require('jsdom');
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;
global.window = window;
global.document = window.document;