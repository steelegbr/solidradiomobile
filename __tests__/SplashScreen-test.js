/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

/**
 * Tests the loading splash screen.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SplashScreen from '../components/SplashScreen';
import { Button } from 'react-native-paper';
import { mount } from 'enzyme';

// Redux

const mockStore = configureStore([]);

// Tests

it('render-not-started', () => {

    // Arrange

    const store = mockStore({
        initialLoad: 'not_started'
    });

    // Act

    const splashScreen = renderer.create(
        <Provider store={store}>
            <SplashScreen />
        </Provider>
    ).toJSON();

    // Assert

    expect(splashScreen).toMatchSnapshot();

});

it('render-started', () => {

    // Arrange

    const store = mockStore({
        initialLoad: 'started'
    });

    // Act

    const splashScreen = renderer.create(
        <Provider store={store}>
            <SplashScreen />
        </Provider>
    ).toJSON();

    // Assert

    expect(splashScreen).toMatchSnapshot();

});

it('render-success', () => {

    // Arrange

    const store = mockStore({
        initialLoad: 'success'
    });

    // Act

    const splashScreen = renderer.create(
        <Provider store={store}>
            <SplashScreen />
        </Provider>
    ).toJSON();

    // Assert

    expect(splashScreen).toMatchSnapshot();

});

it('render-error', () => {

    // Arrange

    const store = mockStore({
        initialLoad: 'error'
    });

    // Act

    const splashScreen = renderer.create(
        <Provider store={store}>
            <SplashScreen />
        </Provider>
    ).toJSON();

    // Assert

    expect(splashScreen).toMatchSnapshot();

});

it('retry-pressed', () => {

    // Arrange

    const store = mockStore({
        initialLoad: 'error'
    });

    const splashScreen = mount(
        <Provider store={store}>
            <SplashScreen />
        </Provider>
    );

    // Act

    splashScreen.find(Button).first().props().onPress();

    // Assert

    const expectedActions = [{ "type": "INITIAL_LOAD_REQUESTED" }];
    expect(store.getActions()).toStrictEqual(expectedActions);

});