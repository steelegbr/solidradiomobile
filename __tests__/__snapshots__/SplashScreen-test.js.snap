// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render-error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7300AE",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/Logo.png",
      }
    }
    style={
      Object {
        "flex": 3,
        "height": undefined,
        "width": "80%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Object {
          "color": "white",
          "flex": 0,
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 20,
          "textAlign": "center",
          "width": "100%",
        },
      ]
    }
  >
    Something went wrong trying to get a classic with every byte.
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 20,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Try Again!
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`render-not-started 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7300AE",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/Logo.png",
      }
    }
    style={
      Object {
        "flex": 3,
        "height": undefined,
        "width": "80%",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Object {
          "color": "white",
          "flex": 0,
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 20,
          "textAlign": "center",
          "width": "100%",
        },
      ]
    }
  >
    We won't be a moment...
  </Text>
</View>
`;

exports[`render-started 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7300AE",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/Logo.png",
      }
    }
    style={
      Object {
        "flex": 3,
        "height": undefined,
        "width": "80%",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Object {
          "color": "white",
          "flex": 0,
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 20,
          "textAlign": "center",
          "width": "100%",
        },
      ]
    }
  >
    We won't be a moment...
  </Text>
</View>
`;

exports[`render-success 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7300AE",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/Logo.png",
      }
    }
    style={
      Object {
        "flex": 3,
        "height": undefined,
        "width": "80%",
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Object {
          "color": "white",
          "flex": 0,
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 20,
          "textAlign": "center",
          "width": "100%",
        },
      ]
    }
  >
    We won't be a moment...
  </Text>
</View>
`;
